### ISSUE TITLE
Support for precompiled binaries / JLLs

### ISSUE THREAD

**Issue by fingolfin** [jw3126/CxxCall.jl#7]
For development it is fine if a package using `CxxCall` compiles some C++ code on the fly.

For deployment to end users, it generally isn't (at least where I come from). We currently use CxxWrap with JLLs to deal with that.

Is it possible to use a JLL with CxxCall? And if so, how would I go about it?

**Comment by jw3126** [comment]
What do you mean by compiling C++ code on the fly? This package does not compile any code for you. It is supposed to work like this:
```julia
...
@cxx function hello_world()
    ...
end
```
Will do two things:
* It allows you to generate C++ source code and allows you to turn that into a shared library in using whatever compile chain you want. Produce a jll package if you like.
* It generates the appropriate ccalls to use that library.

**Comment by fingolfin** [comment]
Thanks, that essentially answers my question. None of that was clear from the README, which just shows a magically looking invocation; nor from the tests.

So this package just generates sources code. Presumably it puts it somewhere (again not clear where; in the tests there is a function `cxxnewfile` taking a filename, so I assume that has to be used). Once I have that source, one needs to figure out setting up a build system and/or a JLL (just like one would have with the hand written C++ glue code needed for using CxxWrap). So from the that perspective, it is quite similar to CxxWrap, except it "magically" generates C++ code from Julia sources instead of me maintaining that manually.

Are all these properties inherited from CxxInterface?

**Comment by jw3126** [comment]
> Are all these properties inherited from CxxInterface?

Yes, both packages generate source code and calls to use that code. Neither package helps with compilation.

**Comment by fingolfin** [comment]
Thank you!
